Core/Object Arts/Dolphin/Base/DelayTest.cls

```diff
@@ -0,0 +1,128 @@
+﻿"Filed out from Dolphin Smalltalk 7"!
+
+DolphinTest subclass: #DelayTest
+	instanceVariableNames: 'delay startTime'
+	classVariableNames: ''
+	poolDictionaries: ''
+	classInstanceVariableNames: ''!
+DelayTest guid: (GUID fromString: '{0b7c6875-1051-4b80-b137-cd1c02754bcd}')!
+DelayTest comment: ''!
+!DelayTest categoriesForClass!Unclassified! !
+!DelayTest methodsFor!
+
+expectDelay: expectedMilliseconds for: nowMicroseconds
+	| elapsed delta |
+	elapsed := nowMicroseconds - startTime.
+	delta := (elapsed - (expectedMilliseconds * 1000)) abs.
+	self assert: delta < 2000
```

Comment:
The variance that can be expected is asymmetric. It should be no more than the resolution shorter (i.e. a 10mS wait won't be less than 9mS), but could be considerably longer.

Reply:
I don't want to allow it to be *too* considerably longer as that might allow an actual bug to slip through, but I've shifted the window so it's +3ms/-1ms.